htlcswitch/link_test.go

```diff
@@ -6490,94 +6490,6 @@ func TestPendingCommitTicker(t *testing.T) {
 	}
 }
 
-// TestShutdownIfChannelClean tests that a link will exit the htlcManager loop
-// if and only if the underlying channel state is clean.
-func TestShutdownIfChannelClean(t *testing.T) {
```

Comment:
TODO: Figure out an alternative test to this, since just deleting this test is unacceptable.

Note for Reviewers: If you have opinions on how I should go about testing this new asynchronous shutdown procedure please reply to this thread with your ideas.

Reply:
Test cases I want to add:

- Ensure that the flush operation itself works. It must block update_adds while flushing
- Ensure that the switch doesn't forward to the link when it is flushing.
- Ensure that inbound shutdown results in immediate outbound shutdown and then puts it into a flushing state